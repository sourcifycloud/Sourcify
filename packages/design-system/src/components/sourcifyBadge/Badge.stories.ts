import type { StoryFn } from '@storybook/vue';
import sourcifyBadge from './Badge.vue';

export default {
	title: 'Atoms/Badge',
	component: sourcifyBadge,
	argTypes: {
		theme: {
			type: 'text',
			options: ['default', 'primary', 'secondary', 'tertiary'],
		},
		size: {
			type: 'select',
			options: ['small', 'medium', 'large'],
		},
	},
};

const Template: StoryFn = (args, { argTypes }) => ({
	props: Object.keys(argTypes),
	components: {
		sourcifyBadge,
	},
	template: '<sourcify-badge v-bind="$props">Badge</sourcify-badge>',
});

export const Badge = Template.bind({});
Badge.args = {};
