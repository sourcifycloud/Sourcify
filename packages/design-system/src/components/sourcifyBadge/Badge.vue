<template>
	<span :class="['sourcify-badge', $style[theme]]">
		<sourcify-text :size="size" :bold="bold" :compact="true">
			<slot></slot>
		</sourcify-text>
	</span>
</template>

<script lang="ts">
import sourcifyText from '../sourcifyText';

import { defineComponent } from 'vue';

export default defineComponent({
	props: {
		theme: {
			type: String,
			default: 'default',
			validator: (value: string) => ['default', 'primary', 'secondary', 'tertiary'].includes(value),
		},
		size: {
			type: String,
			default: 'small',
		},
		bold: {
			type: Boolean,
			default: false,
		},
	},
	components: {
		sourcifyText,
	},
});
</script>

<style lang="scss" module>
.badge {
	display: inline-flex;
	align-items: center;
	padding: var(--spacing-5xs) var(--spacing-4xs);
	border: var(--border-base);
	white-space: nowrap;
}

.default {
	composes: badge;
	border-radius: var(--border-radius-base);
	color: var(--color-text-light);
	border-color: var(--color-text-light);
}

.primary {
	composes: badge;
	padding: var(--spacing-5xs) var(--spacing-3xs);
	border-radius: var(--border-radius-xlarge);
	color: var(--color-foreground-xlight);
	background-color: var(--color-primary);
	border-color: var(--color-primary);
}

.secondary {
	composes: badge;
	border-radius: var(--border-radius-xlarge);
	color: var(--color-secondary);
	background-color: var(--color-secondary-tint-1);
}

.tertiary {
	composes: badge;
	border-radius: var(--border-radius-base);
	color: var(--color-text-light);
	border-color: var(--color-text-light);
	padding: 1px var(--spacing-5xs);
}
</style>
