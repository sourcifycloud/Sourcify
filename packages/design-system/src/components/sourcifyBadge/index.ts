import sourcifyBadge from './Badge.vue';

export default sourcifyBadge;
