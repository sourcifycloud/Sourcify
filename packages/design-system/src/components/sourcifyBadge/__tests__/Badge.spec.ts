import { render } from '@testing-library/vue';
import sourcifyBadge from '../Badge.vue';

describe('components', () => {
	describe('sourcifyBadge', () => {
		describe('props', () => {
			it('should render default theme correctly', () => {
				const wrapper = render(sourcifyBadge, {
					props: {
						theme: 'default',
						size: 'large',
						bold: true,
					},
					slots: {
						default: '<sourcify-text>Default badge</sourcify-text>',
					},
					stubs: ['sourcify-text'],
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
			it('should render secondary theme correctly', () => {
				const wrapper = render(sourcifyBadge, {
					props: {
						theme: 'secondary',
						size: 'medium',
						bold: false,
					},
					slots: {
						default: '<sourcify-text>Secondary badge</sourcify-text>',
					},
					stubs: ['sourcify-text'],
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
			it('should render with default values correctly', () => {
				const wrapper = render(sourcifyBadge, {
					slots: {
						default: '<sourcify-text>A Badge</sourcify-text>',
					},
					stubs: ['sourcify-text'],
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
		});
	});
});
