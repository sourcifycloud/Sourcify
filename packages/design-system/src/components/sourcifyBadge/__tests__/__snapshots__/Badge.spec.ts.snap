// Vitest Snapshot v1

exports[`components > sourcifyBadge > props > should render default theme correctly 1`] = `"<span class=\\"sourcify-badge default\\"><sourcify-text-stub bold=\\"true\\" size=\\"large\\" compact=\\"true\\" tag=\\"span\\"><sourcify-text-stub size=\\"medium\\" tag=\\"span\\">Default badge</sourcify-text-stub></sourcify-text-stub></span>"`;

exports[`components > sourcifyBadge > props > should render secondary theme correctly 1`] = `"<span class=\\"sourcify-badge secondary\\"><sourcify-text-stub size=\\"medium\\" compact=\\"true\\" tag=\\"span\\"><sourcify-text-stub size=\\"medium\\" tag=\\"span\\">Secondary badge</sourcify-text-stub></sourcify-text-stub></span>"`;

exports[`components > sourcifyBadge > props > should render with default values correctly 1`] = `"<span class=\\"sourcify-badge default\\"><sourcify-text-stub size=\\"small\\" compact=\\"true\\" tag=\\"span\\"><sourcify-text-stub size=\\"medium\\" tag=\\"span\\">A Badge</sourcify-text-stub></sourcify-text-stub></span>"`;
